### ISSUE TITLE
Use cloudpickle instead of pickle

### ISSUE THREAD

**Issue by timvink** [koaning/uvtrick#22]
I am writing some unit tests using uvtrick. If you define objects inside a function scope, pickle can't pickle them.

[cloudpickle ](https://github.com/cloudpipe/cloudpickle) "makes it possible to serialize Python constructs not supported by the default pickle module from the Python standard library."

Shall we use that instead? Happy to provide a PR if you want.

**Comment by timvink** [comment]
Here's an example error message you get when not using `cloudpickle`: 

```
Traceback (most recent call last):
  File "/tmp/tmp311h9dlh/pytemp.py", line 55, in <module>
    args, kwargs = cloudpickle.loads(Path('/tmp/tmp311h9dlh/pickled_inputs.pickle').read_bytes())
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'tests'
```

**Comment by timvink** [comment]
I have been using cloudpickle happily in many projects. The only downside is the extra dependency (see the PR)

**Comment by timvink** [comment]
No experience with `dill` yet

**Comment by timvink** [comment]
Closed via #23
